android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -330,9 +330,20 @@ public static View createInputView(InputMethodService inputMethodService) {
   }
 
   public static void onStartInput(EditorInfo attribute, boolean restarting) {
+    int inputType = attribute.inputType;
+
+    // Temporarily disable lm-layer if entering a hidden password field
+    if (currentLexicalModel != null && isHiddenPasswordInputType(inputType)) {
+      deregisterLexicalModel(currentLexicalModel.get(KMKey_LexicalModelID));
+    } else if (currentLexicalModel == null) {
+      // Check if lm-layer needs to be re-enabled
+      HashMap<String, String> kbInfo = getCurrentKeyboardInfo(appContext);
+      String langId = kbInfo.get(KMKey_LanguageID);
+      registerAssociatedLexicalModel(langId);
+    }
```

Comment:
Note that KMW's `com.keyman.text.prediction.ModelManager` class has the property `mayPredict` that was designed for this sort of case; rather than completely dumping the model, it may simply be turned off and on selectively once registered.

Of course, the approach above works too.

Reply:
ah, I'll update to use `mayPredict` since that's better.

Reply:
> ah, I'll update to use `mayPredict`

Well, I tried. The problem is the initial time the system keyboard gets loaded, it calls `registerAssociatedLexicalModel` again and pulls `mayPredict` with whatever the setting is. (default true). I'll keep the code as it is.

Reply:
It sounds like `mayPredict` would be better in theory -- cleaner than a de-reg and re-reg. Is it possible to fix the underlying design issue there?

Reply:
ok, I've updated so `mayPredict` will override to `false` when entering a password field.